### ISSUE TITLE
3.4.1 - Line indented incorrectly false positive

### ISSUE THREAD

**Issue by Dmitrev** [squizlabs/PHP_CodeSniffer#2462]
Starting from v3.4.1 (works in v3.4.0) the linter gives false positive for certain code

Here is some pseudo code that currently triggers the error

```
<?php

namespace App;

class App extends Base
{
    public static function doSomething()
    {
        return cache(
            "key",
             10,
             function () use($arg) {
                 if ($arg === 'something') { //Line indented incorrectly; expected 8 spaces,  found 16
                      do_something();
                 }
             }
        )
    }
}
```

**Comment by webimpress** [comment]
@Dmitrev It might be already fixed - check `master` branch and see also:
https://github.com/squizlabs/PHP_CodeSniffer/issues/2414

I've done it and I am not getting any issues on provided example.

**Comment by gsherwood** [comment]
The provided code snippet doesn't actually generated any errors for me in 3.4.1, although the alignment of the function args needs to be corrected first as the lines after `"key"` are indented 1 extra character.

So I can't 100% confirm your issue is fixed, but it looks like the same basic code layout as #2450, so I think it's probably a duplicate of that. I'm trying to get a 3.4.2 release out this week to correct that, so it would be great if you were able to confirm the current master code fixes your issue as well.

**Comment by binarious** [comment]
I can confirm this issue. Some of my code throws the indentation error with 3.4.1, but not with 3.4.0. 

My test case:
```php
<?php

namespace App;

class App
{
    public function exportToCSV($query, $response)
    {
        $response->setCallback(
            function () use ($response) {
                while ($row = $query->fetch()) {
                    $userValues = array(
                        'test'
                    );
                    if ($i % $batchSize == 0) { // Line indented incorrectly; expected 8 spaces, found 20
                        echo 'test';
                    } // Line indented incorrectly; expected 8 spaces, found 20
                }
            }
        );
    }
}

```

The issue seems to disappear when I remove the following:
```php
$userValues = array(
    'test'
);
```

**Comment by skors** [comment]
Hi,

having the same issue here on 3.4.1, testing with standard=PSR2

```php
<?php

namespace Test;

class TestClass
{
    private function testMethod(ServiceLocatorInterface $serviceManager)
    {
        return new SomeClass(
            [
                'basket_items' => [
                    'hydrate' => function (array $data) use ($serviceManager) {
                        $basketItemService = $serviceManager->get('SomeOtherBasket\Service\BasketItem');

                        if (isset($data['basket_items'])
                            && is_array($data['basket_items'])
                            && count($data['basket_items']) > 0
                        ) {
                            $basketItems = [];
                            foreach ($data['basket_items'] as $basketItem) { // Line indented incorrectly; expected 12 spaces, found 28
                                if ($basketItem instanceof BasketItemModel) { // Line indented incorrectly; expected 16 spaces, found 32
                                    $basketItems[] = $basketItem;
                                } else { // Line indented incorrectly; expected 16 spaces, found 32
                                    $basketItems[] = $basketItemService->find($basketItem);
                                } // Line indented incorrectly; expected 16 spaces, found 32
                            } // Line indented incorrectly; expected 12 spaces, found 28
                        }

                        return $basketItems;
                    },
                ],
            ]
        );
    }
}
```
Results:
```
FILE: ./test.php
--------------------------------------------------------------------------------------------------------------------------
FOUND 5 ERRORS AFFECTING 5 LINES
--------------------------------------------------------------------------------------------------------------------------
 25 | ERROR | [x] Line indented incorrectly; expected 12 spaces, found 28
    |       |     (Generic.WhiteSpace.ScopeIndent.IncorrectExact)
 26 | ERROR | [x] Line indented incorrectly; expected 16 spaces, found 32
    |       |     (Generic.WhiteSpace.ScopeIndent.IncorrectExact)
 28 | ERROR | [x] Line indented incorrectly; expected 16 spaces, found 32
    |       |     (Generic.WhiteSpace.ScopeIndent.IncorrectExact)
 30 | ERROR | [x] Line indented incorrectly; expected 16 spaces, found 32
    |       |     (Generic.WhiteSpace.ScopeIndent.IncorrectExact)
 31 | ERROR | [x] Line indented incorrectly; expected 12 spaces, found 28
    |       |     (Generic.WhiteSpace.ScopeIndent.IncorrectExact)
--------------------------------------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 5 MARKED SNIFF VIOLATIONS AUTOMATICALLY
--------------------------------------------------------------------------------------------------------------------------

Time: 49ms; Memory: 6MB
```
But without the multiline if it works:
```php
<?php

namespace Test;

class TestClass
{
    private function testMethod(ServiceLocatorInterface $serviceManager)
    {
        return new SomeClass(
            [
                'basket_items' => [
                    'hydrate' => function (array $data) use ($serviceManager) {
                        $basketItemService = $serviceManager->get('SomeOtherBasket\Service\BasketItem');

                        if (isset($data['basket_items']) && is_array($data['basket_items']) && (count($data['basket_items']) > 0)) {
                            $basketItems = [];
                            foreach ($data['basket_items'] as $basketItem) {
                                if ($basketItem instanceof BasketItemModel) {
                                    $basketItems[] = $basketItem;
                                } else {
                                    $basketItems[] = $basketItemService->find($basketItem);
                                }
                            }
                        }

                        return $basketItems;
                    },
                ],
            ]
        );
    }
}
```
Result:
```
. 1 / 1 (100%)


Time: 59ms; Memory: 6MB
```

**Comment by gsherwood** [comment]
Closing this issue as a duplicate of #2450. About to prepare a release to include this fix.
